internal/services/automation/automation_job_schedule_resource.go

```diff
@@ -187,41 +187,53 @@ func resourceAutomationJobScheduleRead(d *pluginsdk.ResourceData, meta interface
 	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id, err := jobschedule.ParseJobScheduleID(d.Id())
+	// the jobSchedule ID may be updated by Runbook, so need to get the real id by list API
+	tfID, err := commonids.ParseCompositeResourceID(d.Id(), &schedule.ScheduleId{}, &runbook.RunbookId{})
 	if err != nil {
 		return err
 	}
 
-	resp, err := client.Get(ctx, *id)
+	js, err := GetJobScheduleFromTFID(ctx, client, tfID)
 	if err != nil {
-		if response.WasNotFound(resp.HttpResponse) {
-			d.SetId("")
-			return nil
-		}
-		return fmt.Errorf("making Read request on %s: %+v", *id, err)
+		return err
+	}
+	if js == nil {
+		d.SetId("")
+		return nil
 	}
```

Comment:
In the case where the resource group has been deleted outside of terraform, this will continually give us an error when trying to list the job schedules and it is never marked as gone. We should update this to cover this case.